File: bowling_game.py
class BowlingGame:
    def __init__(self):
        self.rolls = []

    def roll(self, pins):
        self.rolls.append(pins)

    def score(self):
        score = 0
        next_scored_roll = 0

        for _ in range(10):
            frame_score = 0

            if self.rolls[next_scored_roll] == 10:
                frame_score += self.rolls[next_scored_roll]
                frame_score += self.rolls[next_scored_roll + 1]
                frame_score += self.rolls[next_scored_roll + 1]
                next_scored_roll += 1
            elif self.rolls[next_scored_roll] + self.rolls[next_scored_roll + 1] == 10:
                frame_score += self.rolls[next_scored_roll]
                frame_score += self.rolls[next_scored_roll + 1]
                frame_score += self.rolls[next_scored_roll + 2]
                next_scored_roll += 1
                next_scored_roll += 1
            else:
                frame_score += self.rolls[next_scored_roll]
                frame_score += self.rolls[next_scored_roll + 1]
                next_scored_roll += 1
                next_scored_roll += 1

            score += frame_score

        return score

File: javascript/bowlingGame.js
class BowlingGame {
  _rolls = [];

  constructor() {
    this._rolls = [];
  }

  roll(pins) {
    this._rolls.push(pins);
  }

  score() {
    let score = 0;
    let nextScoredRoll = 0;

    for (let i = 0; i < 10; i++) {
      let frameScore = 0;
      if (this._rolls[nextScoredRoll] == 10) {
        frameScore += this._rolls[nextScoredRoll];
        frameScore += this._rolls[nextScoredRoll + 1];
        frameScore += this._rolls[nextScoredRoll + 1];
        nextScoredRoll++;
      } else if (
        this._rolls[nextScoredRoll] + this._rolls[nextScoredRoll + 1] ===
        10
      ) {
        frameScore += this._rolls[nextScoredRoll];
        frameScore += this._rolls[nextScoredRoll + 1];
        frameScore += this._rolls[nextScoredRoll + 2];
        nextScoredRoll += 2;
      } else {
        frameScore += this._rolls[nextScoredRoll];
        frameScore += this._rolls[nextScoredRoll + 1];
        nextScoredRoll += 2;
      }
      score += frameScore;
    }

    return score;
  }
}

module.exports = BowlingGame;
